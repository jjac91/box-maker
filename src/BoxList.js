import React, { useState } from "react";
import NewBoxForm from "./NewBoxForm";
import Box from "./Box";
import { v4 as uuid } from "uuid";

const BoxList = () => {
  const [boxes, setBoxes] = useState([]);

  const addBox = (boxData) => {
    let newBox = { ...boxData, id: uuid() };
    setBoxes((boxes) => [...boxes, newBox]);
  };
  const remove = (id) => {
    setBoxes((boxes) => boxes.filter((box) => box.id !== id));
  };

  const renderBoxes = boxes.map((box) => (
    <Box
      id={box.id}
      color={box.color}
      height={box.height}
      width={box.width}
      key={box.id}
      handleRemove={remove}
    />
  ));

  return (
    <div>
      <NewBoxForm addBox={addBox} />
      {renderBoxes}
    </div>
  );
};

export default BoxList;
