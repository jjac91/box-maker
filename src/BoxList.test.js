import React from "react";
import { fireEvent, render } from "@testing-library/react";
import BoxList from "./BoxList";

function addBox(boxList, height = "50", width = "50", color = "blue") {
  const heightInput = boxList.getByLabelText("Height:");
  const widthInput = boxList.getByLabelText("Width:");
  const colorInput = boxList.getByLabelText("Color:");
  fireEvent.change(colorInput, { target: { value: color } });
  fireEvent.change(widthInput, { target: { value: width } });
  fireEvent.change(heightInput, { target: { value: height } });
  const button = boxList.getByText("Add Box");
  fireEvent.click(button);
}
it("renders without crashing", function () {
  render(<BoxList />);
});

it("matches snapshot", function () {
  const { asFragment } = render(<BoxList />);
  expect(asFragment()).toMatchSnapshot();
});

it("can add a new box", function () {
  const boxList = render(<BoxList />);
  //No boxes created yet
  expect(boxList.queryByText("Remove this box")).not.toBeInTheDocument();

  addBox(boxList);

  const deleteButton = boxList.getByText("Remove this box");
  expect(deleteButton).toBeInTheDocument();
  console.log(deleteButton.previousSibling);
  expect(deleteButton.previousSibling).toHaveStyle(`
  background-color: blue;
  width: 50px;
  height: 50px;
  `);
});
it("can remove a box", function () {
  const boxList = render(<BoxList />);
  addBox(boxList);

  const deleteButton = boxList.getByText("Remove this box");
  fireEvent.click(deleteButton);
  expect(deleteButton).not.toBeInTheDocument();
});
