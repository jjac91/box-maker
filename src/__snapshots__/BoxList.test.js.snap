// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <form>
      <label
        for="color"
      >
        Color:
      </label>
      <input
        id="color"
        name="color"
        type="text"
        value=""
      />
      <label
        for="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        type="text"
        value=""
      />
      <label
        for="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        type="text"
        value=""
      />
      <button
        id="newBoxButton"
      >
        Add Box
      </button>
    </form>
  </div>
</DocumentFragment>
`;
